docs/csharp/tour-of-csharp/types.md

```diff
@@ -0,0 +1,108 @@
+
+### Type parameters
+
+Generic classes define [***type parameters***](programming-guide/generics/index.md). Type parameters are a list of type parameters names enclosed in angle brackets. Type parameters follow the class name. The type parameters can then be used in the body of the class declarations to define the members of the class. In the following example, the type parameters of `Pair` are `TFirst` and `TSecond`:
+
+:::code language="csharp" source="./snippets/shared/Types.cs" ID="DefinePairClass":::
+
+A class type that is declared to take type parameters is called a *generic class type*. Struct, interface, and delegate types can also be generic.
+When the generic class is used, type arguments must be provided for each of the type parameters:
+
+:::code language="csharp" source="./snippets/shared/Types.cs" ID="CreatePairObject":::
+
+A generic type with type arguments provided, like `Pair<int,string>` above, is called a *constructed type*.
+
+### Base classes
+
+A class declaration may specify a base class. Follow the class name and type parameters with a colon and the name of the base class. Omitting a base class specification is the same as deriving from type `object`. In the following example, the base class of `Point3D` is `Point`. From the first example, the base class of `Point` is `object`:
+
+:::code language="csharp" source="./snippets/shared/Types.cs" ID="Create3DPoint":::
+
+A class inherits the members of its base class. Inheritance means that a class implicitly contains almost all members of its base class. A class doesn't inherit the instance and static constructors, and the finalizer. A derived class can add new members to those members it inherits, but it can't remove the definition of an inherited member. In the previous example, `Point3D` inherits the `x` and `y` fields from `Point`, and every `Point3D` instance contains three fields, `x`, `y`, and `z`.
+
+An implicit conversion exists from a class type to any of its base class types. A variable of a class type can reference an instance of that class or an instance of any derived class. For example, given the previous class declarations, a variable of type `Point` can reference either a `Point` or a `Point3D`:
+
+:::code language="csharp" source="./snippets/shared/Types.cs" ID="ImplicitCastToBase":::
+
+## Structs
+
+Classes define types that support inheritance and polymorphism. They enable you to create sophisticated behaviors based on hierarchies of derived classes. By contrast, [***struct***](../language-reference/builtin-types/value-types.md) types are simpler types whose primary purpose is to store data values. Structs can't declare a base type; they implicitly derive from <xref:System.ValueType?displayProperty=nameWithType>. You can't derive other `struct` types from a `struct` type. They're implicitly sealed.
```

Comment:
```suggestion
Classes define types that support inheritance and polymorphism. They enable you to create sophisticated behaviors based on hierarchies of derived classes. By contrast, [***struct***](../language-reference/builtin-types/struct.md) types are simpler types whose primary purpose is to store data values. Structs can't declare a base type; they implicitly derive from <xref:System.ValueType?displayProperty=nameWithType>. You can't derive other `struct` types from a `struct` type. They're implicitly sealed.
```

Reply:
Good change. (Interesting in that I was using the TOC filter, and it didn't bring this article up. I'll check again)

Reply:
@BillWagner I've just checked the TOC filter with the "struct" entry. My observations:
- A lot of articles about con**struct**ors are brought up
- Suggestions look to be sorted by order in TOC. The language reference node is almost the last one (after the programming guide one).
- That makes the "Structure types" article suggestion to be the second from the list bottom. You should scroll down through the brought up suggestions to notice it.

Searching for "structure" in the TOC filter is better.